code/modules/power/supermatter/supermatter.dm

```diff
@@ -186,6 +190,11 @@
 		qdel(src)
 
 /obj/machinery/power/supermatter_shard/process_atmos()
+	if(process_tick >= process_modifier)
```

Comment:
replace all this you've added with:

```
if(!(process_tick & process_modifier - 1))
    return
```
Then put a comment on ```process_modifier``` stating it must be a power of 2.